school_center.py

```diff
@@ -68,27 +71,53 @@ def sort_key(c):
     else: # if there are no centers within given  threshold, return one that is closest
         return [center_to_dict(nearest_center, nearest_distance)]
 
+
 def read_tsv(file_path: str) -> List[Dict[str, str]]:
     data = []
-    with open(file_path, 'r', newline='', encoding='utf-8') as file:
-        reader = csv.DictReader(file, delimiter='\t')
-        for row in reader:
-            data.append(dict(row))
+    try:
+        with open(file_path, 'r', newline='', encoding='utf-8') as file:
+            reader = csv.DictReader(file, delimiter='\t')
+            for row in reader:
+                data.append(dict(row))
+    except FileNotFoundError as e:
+        logging.error(f"Error: File '{file_path}' not found.")
+        sys.exit(1)
+    except PermissionError as e:
+        logging.error(f"Error: Permission denied while accessing file '{file_path}'.")
+        sys.exit(1)
+    except IOError as e:
+        logging.error(f"Error opening or reading file: {file_path}")
+        sys.exit(1)
+    except Exception as e:
+        logging.error(f"Error: An unexpected error occurred while reading file '{file_path}': {e}")
+        sys.exit(1)
     return data
 
 def read_prefs(file_path: str) -> Dict[str, Dict[str, int]]:
     prefs = {}
-    with open(file_path, 'r', newline='', encoding='utf-8') as file:
-        reader = csv.DictReader(file, delimiter='\t')
-        for row in reader:
-            if prefs.get(row['scode']):
-                if prefs[row['scode']].get(row['cscode']):
-                    prefs[row['scode']][row['cscode']] += int(row['pref'])
+    try:
```

Comment:
Since we are at it, could we also add test for it please 💙

Reply:
Sure, but I'd like to know the convention to add tests for this project, should I create a new directory and add tests there or what should I do? Please guide me!

Reply:
test cases are being handled in #4 . 
will merge if we are good with error handling piece.

Reply:
It would always be good to add tests for any changed piece of code one does. So it'd be nice to have the tests for error-handling cases because that's the only thing that has changed here. #4 might need to have this change in place to cover it. However, someone is already doing it, let's merge it and check this through later on to make sure that it is covered.